guides/user/managed-delivery/environment-constraints/index.md

```diff
@@ -0,0 +1,316 @@
+ensure promotion to an environment only occurs during business hours, or quiet times.
+
+**Parameters**
+`windows`: A list of time windows; multiple windows can be specified on a single constraint. The constraint is
+satisfied if the current time matches any single window.
+`tz`: *Optional* The timezone to evaluate the constraint in. This is optional, by default the timezone specified via
+the `default.time-zone` spring configuration property is used, not the JVM or OS timezone. If your organization
+operates in a single timezone, this property should likely be set that timezone.
+
+**Window Parameters**
+`hours`: Supports ranges (i.e. `10-20`) and comma separated lists (i.e. `5,6,7,8`) or a combination
+of the two.
+`days`: Supports aliases (`weekdays` and `weekends`), as well as ranges and comma separated lists of
+days in their java standard (`java.util.Calendar`) short (`mon-thu`) or full (`monday-thursday`) names.
+
+**Example**
+```yaml
+  constraints:
+  - type: allowed-times
+    windows:
+      - days: Monday-Wednesday,Friday
+        hours: 8-16
+      - days: Thursday
+        hours: 14-16
+```
+This allows artifacts to deploy into the environment between 8am-4pm on Monday, Tuesday, Wednesday, and Friday or
+between 2-4pm on Thursday.
+
+### Manual Judgement
+
+The `manual-judgement` constraint prevents an artifact from deploying to an environment without explicit
+approval via an external API client. UI support for approving or rejecting judgements as well as an 
+interactive slack bot are under development but direct use of Spinnaker's REST API is currently required to
+answer a manual judgement.
+ 
+**Parameters**
+
+`timeout`: *(Optional)*; Type: `Duration`; Default Value: `PT1W`
+
+If judgement is not provided within this time (starting when the constraint is initially evaluated for a 
+given environment and artifact version), it is considered rejected.
+ 
+**Example**
+```yaml
+  constraints:
+  - type: manual-judgement
+```
+
+### Pipeline
+
+The `pipeline` constraint launches a pipeline, optionally passing along constraint provided trigger
+parameters. The pipeline execution is periodically monitored and the constraint status derived from
+the final status of the execution. 
+
+**Parameters**
+
+`pipelineId`: Type: `String`
+
+The configuration id of the pipeline to be executed during constraint evaluation. If unknown,
+edit the given pipeline from the Spinnaker UI. From the edit configuration view, the `pipelineId` is the
+final part of the URL.
+
+`retries`: *(Optional)*; Type: `Integer`
+
+If set, pipeline failures will be retried up-to this many times if pipeline
+execution goes terminal for any reason. The constraint evaluation will fail once retries are exhausted.
+
+`parameters`: *(Optional)*; Type: `Map<String, Object>` 
+
+If set, the contents of this map are passed as trigger parameters when executing the pipeline.
```

Comment:
Worth specifying that these can't be spel expressions? I feel like that question will come up.